xarray/core/combine.py

```diff
@@ -540,8 +540,15 @@ def vars_as_keys(ds):
     return tuple(sorted(ds))
 
 
+def _all_unnamed_data_arrays(data_objects):
+    for data_obj in data_objects:
+        if not isinstance(data_obj, DataArray) or data_obj.name is not None:
+            return False
+    return True
+
+
 def combine_by_coords(
```

Comment:
can we also modify `combine_nested` so the two are consistent.

Reply:
I wrote a test `test_combine_nested_unnamed_data_arrays` that passes a list of unnamed `DataArray`s into `combine_nested` and it produces the expected output. Can you clarify what about `combine_nested` you want to be consistent?

Reply:
I think I know what @dcherian meant - at first glance it looks like the `_combine_single_variable_hypercube` refactoring is a change that would also simplify the code in `combine_nested`. But looking more closely I don't think it actually makes sense to do that, does it? It seems about as neat as it can be as is.